src/components/Firestore/Requests/RequestDetails/CodeViewer/index.tsx

```diff
@@ -110,12 +111,26 @@ const RulesCodeViewer: React.FC<Props> = ({
     );
   };
 
+  function renderCallout() {
+    if (!linesOutcome?.length) {
+      return (
+        <Callout type="note">
```

Comment:
Please use `caution` for consistency with Firebase Console (and also the deny icon color).